lib/run-status.js

```diff
@@ -18,10 +18,15 @@ function sum(arr, key) {
 }
 
 function normalizeError(err) {
-	if (!isObj(err)) {
+	if (isObj(err) && 'message' in err && typeof err.message === 'string') {
+		['name', 'stack'].forEach(field => {
+			if (field in err && typeof err[field] !== 'string') {
+				delete err[field];
```

Comment:
Yes, this totally makes sense, still I find fairly to say that errors coming to `normalizeError` are already serialized (as I can see from the [test-worker](https://github.com/avajs/ava/blob/master/lib/test-worker.js#L42) code).
Also thank you a lot for your fast and detailed feedback.